src/components/Card.js

```diff
@@ -1,48 +1,100 @@
+        this._deleteButton.classList.remove("card__delete-button_hidden");
+        this._deleteButton.addEventListener("click", () =>
+            this._handleDeleteClick()
+        );
+    }
+
+
+    _setEventListeners() {
+        this._cardElement.querySelector(".card__like-button").addEventListener('click', () => {
+            this._handleLikeClick();
         })
-        
-        this._cardElement.querySelector(".card__delete-button").addEventListener('click',()=>{
-            this._handleDeleteCard();
+
+        this._cardElement.querySelector(".card__delete-button").addEventListener('click', () => {
+            this._handleDeleteClick();
         })
-        
-         this._cardElement.querySelector(".card__image").addEventListener('click', () =>{
-            this._handleImageClick({name:this._name,link:this._link});
+
+        this._cardElement.querySelector(".card__image").addEventListener('click', () => {
+            this._handleImageClick({ name: this._name, link: this._link });
         });
-         
+
     }
-    _handleLikeIcon(){
+    _handleLikeIcon() {
         this._cardElement.querySelector('.card__like-button').classList.toggle('card__like-active')
     }
-
-    _handleDeleteCard(){
+    
+    removeCard() {
         this._cardElement.remove();
         this._cardElement = null;
     }
 
-    getView(){
-      this._cardElement = this._getTemplate();
-      this._setEventListeners();
-      this._cardElement.querySelector('.card__image').src = this._link
-      this._cardElement.querySelector('.card__label-title').textContent = this._name
-      this._cardElement.querySelector('.card__image').alt = this._name
-      return this._cardElement;
+    setLikes(isLiked) {
+        console.log(isLiked)
+        this._isLiked = isLiked;
+        //this._cardElement.querySelector(".card__like-button").classList.add("card__like-active");
+        this._renderlikes();
+    }
+    
+    _renderlikes() {
+        if (this._isLiked) {
+            this._cardElement.querySelector(".card__like-button").classList.add("card__like-active");
+        } else {
+            this._cardElement.querySelector(".card__like-button").classList.remove("card__like-active");
+        }
+      }
+
+    isLiked() {
+        return this._isLiked;
+    }
+
+    getView() {
+        // this._cardElement = this._getTemplate();
+        // this._setEventListeners();
+        // this._cardElement.querySelector('.card__image').src = this._link
+        // this._cardElement.querySelector('.card__label-title').textContent = this._name
+        // this._cardElement.querySelector('.card__image').alt = this._name
+        // return this._cardElement;
+        this._cardElement = this._getTemplate();
+
+        this._likeButton = this._cardElement.querySelector(".card__like-button");
+        this._deleteButton = this._cardElement.querySelector(".card__delete-button");
+        this._imageElement = this._cardElement.querySelector(".card__image");
```

Comment:
Good job that you've found these variables only once here and defined them as **class fields**